### DISCUSSION TITLE
How to set private registry username password?

### DISCUSSION THREAD

**Discussion by Dele73** [cri-o/cri-o#7404] [Q&A]
I configured /etc/containers/registries.conf file below format. 
But it is still showing error FATA[0000] pulling image: reading manifest *** in ***.com: unauthorized: access to the requested resource is not authorized.

global_auth_file = "/var/lib/kubelet/config.json"
#
[[registry]]
prefix = "registry.k8s.io"
insecure = false
blocked = false
location = "registry.k8s.io"
#
[[registry.mirror]]
location = "***.com"
insecure = true
[[registry.mirror.auth]]
username = "username@2023"
password = "password@2023"

**Comment by Dele73** [comment]
/var/lib/kubelet/config.json it has below content.
{
    "***.com": {
      "auth": "username@2023:password@2023"
    }
  }
}

**Comment by haircommander** [comment]
@QiWang19 @saschagrunert @mtrmac PTAL

**Comment by mtrmac** [comment]
> `[[registry.mirror.auth]]`

That just doesn’t exist.

> `"auth": "username@2023:password@2023"`

… and that field just be base64-encoded.

---

I would rather strongly recommend collecting a debug-level CRI-O log, or at least to follow the project’s bug reporting template.

In particular, the debug-level log warns about things like unrecognized field names; as well as it should allow tracing the search for credentials.
